sys/include/net/nanocoap.h

```diff
@@ -499,6 +499,18 @@ static inline unsigned coap_get_code_raw(const coap_pkt_t *pkt)
     return (unsigned)pkt->hdr->code;
 }
 
+/**
+ * @brief   Get a request's method type
+ *
+ * @param[in]   pkt   CoAP request packet
+ *
+ * @returns     request method type
+ */
+static inline coap_method_t coap_get_method(const coap_pkt_t *pkt)
+{
+    return pkt->hdr->code;
```

Comment:
Would it make sense to check whether the packet is actually a request and not a response? Something along the lines of `(pkt->hdr->code & 0b11100000) == 0`. Otherwise this could return values with are actually not part of the enum `coap_method_t`.

Reply:
True, I'm not sure how to handle this though - slapping and `assert()` on there is a bad idea because the source of `pkt` is a network packet and we don't want to crash on a failed assertion when badly crafted packet arrives that would otherwise gracefully be ignored by not being part of a e.g. `switch` `case`.

Reply:
I see, I don't have any suggestions for this either.

Reply:
How about another enum for `coap_method_t`? `COAP_METHOD_INVALID`?